### PR TITLE
c: Add runnable for main function

### PR DESCRIPTION
**Pull request by talal**
Release Notes:

- Added Runnable for C main function

This tags can then be used in tasks, for example:

```json
[
  {
    "label": "Run ${ZED_STEM}",
    "command": "gcc",
    "args": [
      "$ZED_FILE",
      "-o",
      "${ZED_DIRNAME}/${ZED_STEM}.out",
      "&&",
      "${ZED_DIRNAME}/${ZED_STEM}.out"
    ],
    "tags": ["c-main"]
  }
]

```

### CODE REVIEW

**Comment by talal** [comment]
@cla-bot check

**Review by osiewicz — APPROVED** [review]
Thanks!

File: crates/languages/src/c/runnables.scm [hunk 1]
```diff
@@ -0,0 +1,20 @@
+; Tag all function definitions
+(
+  (function_definition
```
[the remark this answers is not in the document]
Comment (talal) [reply crates/languages/src/c/runnables.scm]: Makes sense. I've removed the `c-function` tag so now there's only `c-main` similar to how there is `go-main` or `rust-main`.

**Comment by danielzsh** [comment]
Are there plans to introduce a `cpp-main` tag for C++ as well?

**Comment by talal** [comment]
Me, personally, that would be a no since I don't use C++.

Maybe someone else will make a PR for it.

**Pull request merged**